src/server/admin/server/api_server.go

```diff
@@ -257,6 +263,87 @@ func (a *apiServer) Extract(request *admin.ExtractRequest, extractServer admin.A
 			}
 		}
 	}
+	if !request.NoEnterprise {
+		state, err := pachClient.Enterprise.GetState(pachClient.Ctx(), &enterprise.GetStateRequest{})
+		if err != nil {
+			return err
+		}
+		// Don't write out expired or missing enterprise keys
+		if state.State == enterprise.State_ACTIVE {
+			if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{ActivateEnterprise: &enterprise.ActivateRequest{ActivationCode: state.ActivationCode}}}); err != nil {
+				return err
+			}
+		} else {
+			logrus.Warnf("Enterprise license state: %v, not extracting", state.State)
+		}
+	}
+
+	if !request.NoAuth {
+		config, err := pachClient.GetConfiguration(pachClient.Ctx(), &auth.GetConfigurationRequest{})
+		if err == nil {
+			// Activate authentication as `robot:extractrestore`. This token will be used by the Restore process for subsequent operations.
+			if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{ActivateAuth: &auth.ActivateRequest{Subject: "robot:extractrestore"}}}); err != nil {
```

Comment:
This has the slightly gross property that every time we extract/restore we end up creating another token in etcd and throwing it away. Because these have no TTL they could accumulate over time. Realistically doesn't seem like a huge concern, we could always filter tokens for this user out of extracts in the future, or delete the token at the end of the restore.

Reply:
I'm still trying to understand the mechanics here a bit, but IIUC this creates the token during `Restore()`, and my expectation is that a user would only call restore once (otherwise I think they're creating repos and pipelines on top of existing copies of the same repos and pipelines, which I think errors). Basically, I agree it's not a huge deal if we do leak tokens, and I actually think even that's not a risk :+1:

Reply:
This token is a robot token so it's extracted and restored every time the user upgrades. So if they do an extract/restore for 2.0, then 2.1, etc. each of those would call Activate again and create a new robot token which would be included in subsequent extracts.

Like I said, not the end of the world, and we could not extract these in the future as well.

Reply:
oooh, got it. Thanks!